src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -202,24 +224,78 @@ export class ElasticsearchConfig {
     this.password = rawConfig.password;
     this.customHeaders = rawConfig.customHeaders;
 
-    const certificateAuthorities = Array.isArray(rawConfig.ssl.certificateAuthorities)
-      ? rawConfig.ssl.certificateAuthorities
-      : typeof rawConfig.ssl.certificateAuthorities === 'string'
-      ? [rawConfig.ssl.certificateAuthorities]
-      : undefined;
+    const { alwaysPresentCertificate, verificationMode } = rawConfig.ssl;
 
-    this.ssl = {
-      ...rawConfig.ssl,
-      certificateAuthorities,
+    let key: string | undefined;
+    let keyPassphrase: string | undefined;
+    let certificate: string | undefined;
+    let certificateAuthorities: string[] | undefined;
+    const addCAs = (ca: string[] | undefined) => {
+      if (ca && ca.length) {
+        certificateAuthorities = certificateAuthorities ? certificateAuthorities.concat(ca) : ca;
+      }
     };
 
-    if (this.username === 'elastic' && log !== undefined) {
-      // logger is optional / not used during tests
-      // TODO: logger can be removed when issue #40255 is resolved to support deprecations in NP config service
-      log.warn(
-        `Setting the elasticsearch username to "elastic" is deprecated. You should use the "kibana" user instead.`,
-        { tags: ['deprecation'] }
+    if (rawConfig.ssl.keystore?.path) {
+      const { key: k, cert, ca } = readPkcs12Keystore(
+        rawConfig.ssl.keystore.path,
+        rawConfig.ssl.keystore.password
       );
+      if (!k && !cert) {
+        log.warn(
+          `Did not find key or certificate in keystore; mutual TLS authentication is disabled.`
```

Comment:
I logged a warning here because of how I handled the case where a key is defined without a cert (see comment below). However, since this only applies to PKCS12 key stores, I agree we can throw an error here instead. I'll change it.